src/MIDebugEngine/Engine.Impl/DebugUnixChildProcess.cs

```diff
@@ -0,0 +1,260 @@
+                await _process.MICommandFactory.Catch("fork");
+                await _process.MICommandFactory.Catch("vfork");
+                _state = State.Enabled;
+                _newpid = 0;
+            }
+            else if (!enable)
+            {
+                if (_forkBp != null)
+                {
+                    await _process.MICommandFactory.BreakDelete(_forkBp);
+                    _forkBp = null;
+                }
+                if (_vforkBp != null)
+                {
+                    await _process.MICommandFactory.BreakDelete(_vforkBp);
+                    _vforkBp = null;
+                }
+                _state = State.Init;
+            }
+        }
+
+        public async Task SetBreakAtMain()
+        {
+            Results results = await _process.MICommandFactory.BreakInsert("main", null);
+            var bkpt = results.Find("bkpt");
+            if (bkpt is ValueListValue)
+            {
+                _mainBreak = ((ValueListValue)bkpt).Content[0].FindString("number");
+            }
+            else
+            {
+                _mainBreak = bkpt.TryFindString("number");
+            }
+        }
+
+        public async Task<bool> Stopped(Results results)
+        {
+            string reason = null;
+
+            switch (_state)
+            {
+                case State.Enabled:
+                    reason = results.TryFindString("reason");
+                    if (reason == "fork")
+                    {
+                        int threadId = results.FindInt("thread-id");
+                        _newpid = results.FindInt("newpid");
+                        _state = State.AtFork;
+                        await _process.Step(threadId, VisualStudio.Debugger.Interop.enum_STEPKIND.STEP_OUT, VisualStudio.Debugger.Interop.enum_STEPUNIT.STEP_LINE);
+                        break;
+                    }
+                    else if (reason == "vfork")
+                    {
+                        int threadId = results.FindInt("thread-id");
+                        _newpid = results.FindInt("newpid");
+                        await _process.MICommandFactory.Set("schedule-multiple", "on");
+                        await _process.MICommandFactory.Catch("exec", onlyOnce: true);
+                        var thread = await _process.ThreadCache.GetThread(threadId);
+                        _state = State.AtVfork;
+                        await _process.Continue(thread);
+                        break;
+                    }
+                    return false;
+                case State.AtFork:
+                    await ProcessChild();
+                    break;
+                case State.AtVfork:
+                    if ("exec" == results.TryFindString("reason"))
+                    {
+                        await _process.MICommandFactory.Set("schedule-multiple", "off");
+                        await RunChildToMain();
```

Comment:
The process doesn't handle the SIGSTOP correctly (if just ignores it) when the process is at the start of program (after exec). No idea why. I had to let it run some code before it would correctly respond to the SIGSTOP. Perhaps there is an earlier stopping point that would work too, but "main" was a well known symbols that would work in most cases.